<h1>ACTIVITY 2 - UK ACCIDENT ROAD </h1>
<hr>
<h2>Analyst: Maricel Maribojoc</h2>

<h2>1. Import necessarry libraries.</h2>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h2>2. Convert Dataset to DataFrame</h2>

In [2]:
road_accident = pd.read_csv('datasets\\uk_road_accident.csv')

In [3]:
road_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h2>3. Check DataFrame Information</h2>

In [4]:
road_accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<h2>4. Check Basic Descriptive Statistics</h2>

In [5]:
road_accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

<h2>5. Call Each Column Individually</h2>

In [6]:
road_accident.columns

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type'],
      dtype='object')

In [7]:
road_accident['Index']

0         200701BS64157
1         200701BS65737
2         200701BS66127
3         200701BS66128
4         200701BS66837
              ...      
660674    201091NM01760
660675    201091NM01881
660676    201091NM01935
660677    201091NM01964
660678    201091NM02142
Name: Index, Length: 660679, dtype: object

In [8]:
road_accident['Accident_Severity']

0         Serious
1         Serious
2         Serious
3         Serious
4          Slight
           ...   
660674     Slight
660675     Slight
660676     Slight
660677    Serious
660678    Serious
Name: Accident_Severity, Length: 660679, dtype: object

In [9]:
road_accident['Accident Date']

0           5/6/2019
1           2/7/2019
2         26-08-2019
3         16-08-2019
4           3/9/2019
             ...    
660674    18-02-2022
660675    21-02-2022
660676    23-02-2022
660677    23-02-2022
660678    28-02-2022
Name: Accident Date, Length: 660679, dtype: object

In [10]:
road_accident['Latitude']

0         51.506187
1         51.495029
2         51.517715
3         51.495478
4         51.488576
            ...    
660674    57.374005
660675    57.232273
660676    57.585044
660677    57.214898
660678    57.575210
Name: Latitude, Length: 660679, dtype: float64

In [11]:
road_accident['Light_Conditions']

0               Darkness - lights lit
1                            Daylight
2         Darkness - lighting unknown
3                            Daylight
4               Darkness - lights lit
                     ...             
660674                       Daylight
660675         Darkness - no lighting
660676                       Daylight
660677         Darkness - no lighting
660678                       Daylight
Name: Light_Conditions, Length: 660679, dtype: object

In [12]:
road_accident['District Area']

0         Kensington and Chelsea
1         Kensington and Chelsea
2         Kensington and Chelsea
3         Kensington and Chelsea
4         Kensington and Chelsea
                   ...          
660674                  Highland
660675                  Highland
660676                  Highland
660677                  Highland
660678                  Highland
Name: District Area, Length: 660679, dtype: object

In [13]:
road_accident['Longitude']

0        -0.209082
1        -0.173647
2        -0.210215
3        -0.202731
4        -0.192487
            ...   
660674   -3.467828
660675   -3.809281
660676   -3.862727
660677   -3.823997
660678   -3.895673
Name: Longitude, Length: 660679, dtype: float64

In [14]:
road_accident['Number_of_Casualties']

0         1
1         1
2         1
3         1
4         1
         ..
660674    2
660675    1
660676    1
660677    1
660678    1
Name: Number_of_Casualties, Length: 660679, dtype: int64

In [15]:
road_accident['Number_of_Vehicles']

0         2
1         2
2         3
3         4
4         2
         ..
660674    1
660675    1
660676    3
660677    2
660678    1
Name: Number_of_Vehicles, Length: 660679, dtype: int64

In [16]:
road_accident['Road_Surface_Conditions']

0                  Dry
1          Wet or damp
2                  Dry
3                  Dry
4                  Dry
              ...     
660674             Dry
660675    Frost or ice
660676    Frost or ice
660677     Wet or damp
660678     Wet or damp
Name: Road_Surface_Conditions, Length: 660679, dtype: object

In [17]:
road_accident['Road_Type']

0         Single carriageway
1         Single carriageway
2                        NaN
3         Single carriageway
4                        NaN
                 ...        
660674    Single carriageway
660675    Single carriageway
660676    Single carriageway
660677    Single carriageway
660678      Dual carriageway
Name: Road_Type, Length: 660679, dtype: object

In [18]:
road_accident['Urban_or_Rural_Area']

0         Urban
1         Urban
2         Urban
3         Urban
4         Urban
          ...  
660674    Rural
660675    Rural
660676    Rural
660677    Rural
660678    Rural
Name: Urban_or_Rural_Area, Length: 660679, dtype: object

In [19]:
road_accident['Weather_Conditions']

0            Fine no high winds
1         Raining no high winds
2                           NaN
3            Fine no high winds
4                           NaN
                  ...          
660674       Fine no high winds
660675       Fine no high winds
660676       Fine no high winds
660677       Fine no high winds
660678    Snowing no high winds
Name: Weather_Conditions, Length: 660679, dtype: object

In [20]:
road_accident['Vehicle_Type']

0                                          Car
1                                          Car
2                        Taxi/Private hire car
3         Bus or coach (17 or more pass seats)
4                                Other vehicle
                          ...                 
660674                                     Car
660675                                     Car
660676                                     Car
660677                   Motorcycle over 500cc
660678                                     Car
Name: Vehicle_Type, Length: 660679, dtype: object

<h2>6. Check for Null Values</h2>

In [21]:
road_accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h2>7. Fill Null Values with Statistical Values</h2>

In [22]:
road_accident['Latitude'] = road_accident['Latitude'].fillna(road_accident['Latitude'].mean())
road_accident['Longitude'] = road_accident['Longitude'].fillna(road_accident['Longitude'].mean())
road_accident['Road_Surface_Conditions'] = road_accident['Road_Surface_Conditions'].fillna('unaccounted')
road_accident['Road_Type'] = road_accident['Road_Type'].fillna('unaccounted')
road_accident['Urban_or_Rural_Area'] = road_accident['Urban_or_Rural_Area'].fillna(road_accident['Urban_or_Rural_Area'].mode()[0])
road_accident['Weather_Conditions'] = road_accident['Weather_Conditions'].fillna('unaccounted')

In [23]:
road_accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h2>8. Check Column Data Types</h2>

In [24]:
road_accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660679 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660679 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  660679 non-null  object 
 10  Road_Type                660679 non-null  object 
 11  Urban_or_Rural_Area      660679 non-null  object 
 12  Weather_Conditions       660679 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [25]:
#road_accident['Number_of_Casualties'] = road_accident['Number_of_Casualties'].astype('category')
#road_accident['Number_of_Vehicles'] = road_accident['Number_of_Vehicles'].astype('category')
road_accident['Index'] = road_accident['Index'].astype('category')
road_accident['Accident_Severity'] = road_accident['Accident_Severity'].astype('category')
road_accident['Accident Date'] = road_accident['Accident Date'].astype('category')
road_accident['Light_Conditions'] = road_accident['Light_Conditions'].astype('category')
road_accident['District Area'] = road_accident['District Area'].astype('category')
road_accident['Road_Surface_Conditions'] = road_accident['Road_Surface_Conditions'].astype('category')
road_accident['Road_Type'] = road_accident['Road_Type'].astype('category')
road_accident['Urban_or_Rural_Area'] = road_accident['Urban_or_Rural_Area'].astype('category')
road_accident['Weather_Conditions'] = road_accident['Weather_Conditions'].astype('category')
road_accident['Vehicle_Type'] = road_accident['Vehicle_Type'].astype('category')

In [26]:
road_accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

<h1>Clearing any Incostencies with the Data Set</h1>

In [27]:
road_accident['Accident Date'] = road_accident['Accident Date'].str.strip()
road_accident['Accident Date'] = road_accident['Accident Date'].astype('str')
road_accident['Accident Date'] = road_accident['Accident Date'].str.replace('/', '-')
road_accident['Accident Date'] = pd.to_datetime(road_accident['Accident Date'], dayfirst = True, errors = 'coerce')

<h1>Adjusting Data Types</h1>

In [28]:
road_accident.dtypes

Index                            category
Accident_Severity                category
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                 category
District Area                    category
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions          category
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions               category
Vehicle_Type                     category
dtype: object

<h1>Extracting date information using pandas date time</h1>

In [29]:
road_accident['Year'] = road_accident['Accident Date'].dt.year
road_accident['Month'] = road_accident['Accident Date'].dt.month
road_accident['Day'] = road_accident['Accident Date'].dt.day
road_accident['DayOfWeek'] = road_accident['Accident Date'].dt.dayofweek

<hr>
<h1>QUESTIONS AND INSIGHTS</h1>
<hr>

In [30]:
road_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2              unaccounted               Urban            unaccounted   
3       Single carriageway               Urban     Fine no high winds   
4              unaccounted               Urban            unaccounted   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  Year  Month  Day  DayOfWeek  
0                                        Car  2019      6    5          2  
1                                        Car  2019      7    2          1  
2                      Taxi/Private hire car  2019      8   26          0  
3       Bus or coach (17 or more pass seats)  2019      8   16          4  
4                              Other vehicle  2019      9    3          1  
...                                      ...   ...    ...  ...  

In [31]:
road_accident.columns

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type', 'Year', 'Month', 'Day', 'DayOfWeek'],
      dtype='object')

<h2>1. What is the most common accident severity?</h2>

In [32]:
road_accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>INSIGHT</h3>
<p>The most common accident severity based on the given data is slight.</p>

<h2>2. What is the distribution of accident severity that is recorded? </h2>

In [33]:
road_accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>INSIGHT</h3>
<p>Birmingham district recorded a total of 13491 accident.</p>

<h2>3. Which district area has the lowest number of accidents overall?</h2>

In [34]:
road_accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h3>INSIGHT</h3>
<p>Based on the result above, the district that has the lowest number of accidents is Clackmannashire.</p>

<h2>4. Which road type has the lowest average number of casualties?</h2>

In [35]:
lowest_casualty_road_type = road_accident.groupby('Road_Type')['Number_of_Casualties'].mean()

In [36]:
lowest_casualty_road_type

Road_Type
Dual carriageway      1.477279
One way street        1.192713
Roundabout            1.274891
Single carriageway    1.344666
Slip road             1.423661
unaccounted           1.248230
Name: Number_of_Casualties, dtype: float64

<h3>INSIGHT</h3>
<p>The roundabout road type has recorded the lowest average number of casualties among all the road types.</p>

<h2>5. What is the distribution of road type for accidents with 10 or more casualties?</h2>

In [37]:
road_accident[road_accident['Number_of_Casualties'] >= 10]['Road_Type'].value_counts()

Road_Type
Single carriageway    125
Dual carriageway       77
Slip road               8
Roundabout              7
One way street          4
unaccounted             1
Name: count, dtype: int64

<h3>INSIGHT</h3>
<p>Single carriageway road type recorded more casualties compared to other road types.</p>

<h2>6. What is the total number of casualties for each district area?</h2>

In [38]:
total_casualties_by_district = road_accident.groupby('District Area')['Number_of_Casualties'].sum()

In [39]:
total_casualties_by_district

District Area
Aberdeen City    1508
Aberdeenshire    2529
Adur              817
Allerdale        1663
Alnwick           351
                 ... 
Wychavon         1943
Wycombe          2442
Wyre             1706
Wyre Forest      1340
York             2453
Name: Number_of_Casualties, Length: 422, dtype: int64

<h3>INSIGHT</h3>
<p>The results above shows the total numbers of casualties per district area. </p>

In [40]:
road_accident['Latitude'].max()

np.float64(60.757544)

<h2>7. What is the most common vehicle type involved in accidents?</h2>

In [41]:
most_common_vehicle = road_accident['Vehicle_Type'].mode()[0]

In [42]:
most_common_vehicle

'Car'

<h3>INSIGHT</h3>
<p>The most common vehicle type that is involved in accidents is car.</p>

<h2>8. What is the average number of casualties for each accident severity level?</h2>

In [43]:
avg_casualties_by_severity = road_accident.groupby('Accident_Severity')['Number_of_Casualties'].mean()

In [44]:
avg_casualties_by_severity

Accident_Severity
Fatal      1.903129
Serious    1.467280
Slight     1.331402
Name: Number_of_Casualties, dtype: float64

<h3>INSIGHT</h3>
<p>Based on the results shown above, fatal holds the most number of casualties compared to seriouus and slight accident severity level.</p>

<h2>9. Perform a correlation analysis between number of casualties and number of vehicles.</h2>

In [45]:
corr_cas_v = road_accident['Number_of_Casualties'].corr(road_accident['Number_of_Vehicles'])

In [46]:
corr_cas_v

np.float64(0.2288888612692756)

<h3>INSIGHT</h3>
<p>Shown above is the correlation analysis between number of casualties and number of vehicles.</p>

<h2>10. What are the top 5 most common vehicle types involved in accidents?</h2>

In [47]:
top_vehicle_types = road_accident['Vehicle_Type'].value_counts().head(5)

In [48]:
top_vehicle_types

Vehicle_Type
Car                                     497992
Van / Goods 3.5 tonnes mgw or under      34160
Bus or coach (17 or more pass seats)     25878
Motorcycle over 500cc                    25657
Goods 7.5 tonnes mgw and over            17307
Name: count, dtype: int64

<h3>INSIGHT</h3>
<p>These are the top 5 most common vehicle types that are usually involved in acccidents that might require more attention in terms of safety or vehicle inspections.</p>

<h2>11.  How many accidents occurred during Darkness - lights lit conditions?</h2>

In [49]:
dark_lights_lit_accidents = road_accident[road_accident['Light_Conditions'] == 'Darkness - lights lit'].value_counts().sum()

In [50]:
dark_lights_lit_accidents

np.int64(129335)

<h3>INSIGHT</h3>
<p>A total of 129335 accidents occured during darkness - light lit conditions. It seems that street lighting plays a significant role in accident prevention.</p>

<h2>12. What is the avergae casualties per year?</h2>

In [51]:
avg_casualties_by_year = road_accident.groupby('Accident Date')['Number_of_Casualties'].mean()

In [52]:
avg_casualties_by_year

Accident Date
2019-01-01    1.474627
2019-01-02    1.364865
2019-01-03    1.523929
2019-01-04    1.365432
2019-01-05    1.319101
                ...   
2022-12-27    1.486034
2022-12-28    1.513812
2022-12-29    1.415888
2022-12-30    1.415205
2022-12-31    1.357513
Name: Number_of_Casualties, Length: 1461, dtype: float64

<h3>INSIGHT</h3>
<p>The result above shows how the number of casualties increase or decrease each year.</p>

<h2>13. What is the most common light conditions for fatal accidents?</h2>

In [53]:
road_accident[road_accident['Accident_Severity'] == 'Fatal']['Light_Conditions'].mode()[0]

'Daylight'

<h3>INSIGHT</h3>
<p>Fatal accidents usually occur during daylight.</p>

In [54]:
road_accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2              unaccounted               Urban            unaccounted   
3       Single carriageway               Urban     Fine no high winds   
4              unaccounted               Urban            unaccounted   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  Year  Month  Day  DayOfWeek  
0                                        Car  2019      6    5          2  
1                                        Car  2019      7    2          1  
2                      Taxi/Private hire car  2019      8   26          0  
3       Bus or coach (17 or more pass seats)  2019      8   16          4  
4                              Other vehicle  2019      9    3          1  
...                                      ...   ...    ...  ...  

<h2>14. Compare the average number of casualties in the Kensington and Chelsea district area to the Highland district area.</h2>

In [55]:
k_c_avg = np.round(road_accident[road_accident['District Area'] == 'Kensington and Chelsea']['Number_of_Casualties'].mean() , 1)

In [56]:
k_c_avg

np.float64(1.1)

In [57]:
highland_avg = np.round(road_accident[road_accident['District Area'] == 'Highland']['Number_of_Casualties'].mean() , 1)

In [58]:
highland_avg

np.float64(1.5)

<h3>INSIGHT</h3>
<p>The Highland district area had a high number of casualties, with an average of 1.5, which is much higher compared to the number of casualties in the district area of Kensington and Chelsea.</p>

<h2>15. What is the average number of casualties for accidents on a Single carriageway with Fine no high winds?
</h2>

In [59]:
np.round(road_accident[(road_accident['Road_Type'] == 'Single carriageway') & (road_accident['Weather_Conditions'] == 'Fine no high winds')]['Number_of_Casualties'].mean() , 1)

np.float64(1.3)

<h3>INSIGHT</h3>
<p>The average number of casualties for accidents on a Single carriageway with Fine no high winds is 1.3</p>

<h2>16. What is the average number of vehicles involved in accidents based on road surface conditions? </h2>

In [60]:
road_accident.groupby('Road_Surface_Conditions')['Number_of_Vehicles'].mean()

Road_Surface_Conditions
Dry                     1.841792
Flood over 3cm. deep    1.610619
Frost or ice            1.640006
Snow                    1.746859
Wet or damp             1.829161
unaccounted             1.742424
Name: Number_of_Vehicles, dtype: float64

<h3>INSIGHT</h3>
<p>The results above shows if certain road surface conditions lead to multi-vehicle collisions.</p>

<h2>17. How many casualties involve is Taxi/Private hire car?</h2>

In [61]:
taxi_casualties = road_accident[road_accident['Vehicle_Type'] == 'Taxi/Private hire car']['Number_of_Casualties'].sum()

In [62]:
taxi_casualties

np.int64(18195)

<h3>INSIGHT</h3>
<p>There are 18195 casualties involve in Taxi/Private hire car.</p>

<h2>18. Which vehicle type is most frequently involved in fatal accidents?</h2>

In [63]:
road_accident[road_accident['Accident_Severity'] == 'Fatal']['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      6577
Van / Goods 3.5 tonnes mgw or under       467
Motorcycle over 500cc                     339
Bus or coach (17 or more pass seats)      325
Goods 7.5 tonnes mgw and over             216
Motorcycle 125cc and under                189
Taxi/Private hire car                     155
Motorcycle over 125cc and up to 500cc     105
Motorcycle 50cc and under                  95
Other vehicle                              70
Goods over 3.5t. and under 7.5t            67
Minibus (8 - 16 passenger seats)           29
Agricultural vehicle                       21
Pedal cycle                                 6
Data missing or out of range                0
Ridden horse                                0
Name: count, dtype: int64

<h3>INSIGHT</h3>
<p>Car accident is the type of vehicle that is most frequently involved in fatal accidents. </p>

<h2>19. What is the total number of casualties?</h2>

In [64]:
road_accident['Number_of_Casualties'].sum()

np.int64(896568)

<h2>INSIGHT</h2>
<p>The total number of casualties recorded a total of 896568 from 2019-2022.<p>

<h2>20. How many casualties involved in bus or coach?</h2>

In [65]:
bus_casualties = road_accident[road_accident['Vehicle_Type'] == 'Bus or coach (17 or more pass seats)']['Number_of_Casualties'].value_counts().sum()

In [66]:
bus_casualties

np.int64(25878)

<h3>INSIGHT</h3>
<p>Shown above is the result of casualties involving bus or coach.</p>

<h2>21. </h2>